libsolidity/interface/AssemblyStack.cpp

```diff
@@ -84,7 +84,7 @@ bool AssemblyStack::parseAndAnalyze(std::string const& _sourceName, std::string
 
 void AssemblyStack::optimize()
 {
-	solAssert(m_language != Language::Assembly, "Optimization requested for loose assembly.");
+	solAssert(m_language == Language::StrictAssembly, "Optimization requested for either loose assembly or YUL.");
```

Comment:
Is this change required because the optimiser suite doesn't work with yul?

Reply:
And ideally this should be changed from assertion to `solUnimplemented` (see how `assemble` works).

Reply:
There are some optimizer stages that introduce EVM opcodes (`eq` for example). Some other stages do not take yul types into account properly. My suggestion would be to fix that once we have a wasm backend prototype.

Reply:
> Is this change required because the optimiser suite doesn't work with yul?

My understanding is that the optimizer suite for yul is not fully implemented yet, so we run into known errors if we try to fuzz it.

Reply:
@axic I agree, going to change `solAssert` to `solUnimplemented`